sdk/core/azure-core/src/main/java/com/azure/core/implementation/ReflectionSerializable.java

```diff
@@ -241,7 +179,34 @@ public static Object deserializeAsJsonSerializable(Class<?> jsonSerializable, by
      * @return Whether {@code bodyContentClass} can be used as {@code XmlSerializable}.
      */
     public static boolean supportsXmlSerializable(Class<?> bodyContentClass) {
-        return XML_SERIALIZABLE_SUPPORTED && XML_SERIALIZABLE.isAssignableFrom(bodyContentClass);
+        if (FROM_XML_CACHE.containsKey(bodyContentClass)) {
+            return true;
+        }
+
+        if (!XmlSerializable.class.isAssignableFrom(bodyContentClass)) {
+            return false;
+        }
```

Comment:
Isn't it sufficient to check whether the class is implementing XmlSerializable? Do we need to check if fromXml and toXml methods are present?

Reply:
This is done to handle cases where a class extends a class implementing XmlSerializable but it doesn't implement the interface methods.